lib/sync/syncer_test.go

```diff
@@ -0,0 +1,110 @@
+package sync
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"boscoin.io/sebak/lib/block"
+	"boscoin.io/sebak/lib/common"
+	"boscoin.io/sebak/lib/network"
+	"boscoin.io/sebak/lib/storage"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type SyncerTestContext struct {
+	t         *testing.T
+	st        *storage.LevelDBBackend
+	syncer    *Syncer
+	tickC     chan time.Time
+	syncInfoC chan *SyncInfo
+}
+
+func TestSyncerSetSyncTarget(t *testing.T) {
+	fn := func(tctx *SyncerTestContext) {
+		var (
+			ctx    = context.Background()
+			syncer = tctx.syncer
+			infoc  = tctx.syncInfoC
+		)
+
+		{
+			bk := block.TestMakeNewBlock([]string{})
+			bk.Height = uint64(1)
+			bk.Save(tctx.st)
+		}
+
+		var (
+			height    uint64   = 10
+			nodeAddrs []string = []string{"a", "b"}
+		)
+
+		syncer.validator = &mockValidator{
+			validateFunc: func(ctx context.Context, si *SyncInfo) error {
+				assert.Equal(t, len(si.NodeAddrs), 2)
+				infoc <- si
+				return nil
+			},
+		}
+
+		go func() {
+			syncer.Start()
+		}()
+
+		syncer.SetSyncTargetBlock(ctx, height, nodeAddrs)
+
+		var heights []uint64
+		for si := range infoc {
+			heights = append(heights, si.BlockHeight)
+			if len(heights) >= 9 {
+				close(infoc)
+			}
+		}
+		require.Equal(t, len(heights), 9)
+	}
+	SyncerTest(t, fn)
+}
+
+func SyncerTest(t *testing.T, fn func(*SyncerTestContext)) {
+	st := storage.NewTestStorage()
```

Comment:
Hum, I get the intent here, but sebak should never start without a genesis block, hence this should be `InitTestBlockchain`